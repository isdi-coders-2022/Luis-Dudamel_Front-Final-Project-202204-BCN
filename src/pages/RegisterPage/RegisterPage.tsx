import RegisterForm from "../../components/RegisterForm/RegisterForm";
import RegisterPageStyled from "./RegisterPageStyle";

const RegisterPage = () => {
  return (
    <RegisterPageStyled>
      <RegisterForm />
    </RegisterPageStyled>
  );
};

export default RegisterPage;
