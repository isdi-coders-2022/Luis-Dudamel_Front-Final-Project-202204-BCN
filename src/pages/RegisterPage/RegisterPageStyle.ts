import styled from "styled-components";

const RegisterPageStyled = styled.div`
  button {
    background-color: #2a9d8f;
    width: 100px;
    border-radius: 20px;
  }
  .avatars-container {
    display: flex;
  }
  h1 {
    padding-top: 30px;
  }
`;
export default RegisterPageStyled;
