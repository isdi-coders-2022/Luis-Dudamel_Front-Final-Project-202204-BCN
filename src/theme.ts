import { createTheme } from "@mui/material/styles";
import { red } from "@mui/material/colors";

const theme = createTheme({
  typography: {
    fontFamily: ["Outfit"].join(","),
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: "none",
          "&:hover": {
            backgroundColor: "#2a9d8f",
          },
        },
      },
    },
  },
  palette: {
    background: {
      default: "#ffc027",
    },
    primary: {
      main: "#264653",
    },
    secondary: {
      main: "#19857b",
    },
    error: {
      main: red.A400,
    },
  },
});

export default theme;
