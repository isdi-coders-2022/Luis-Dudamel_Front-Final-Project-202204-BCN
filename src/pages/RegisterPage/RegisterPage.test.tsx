import { ThemeProvider } from "@mui/material";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import TestRenderer from "react-test-renderer";
import theme from "../../theme";
import RegisterPage from "./RegisterPage";
import RegisterPageStyled from "./RegisterPageStyled";

describe("Given a RegisterPage component page", () => {
  describe("When it's invoked", () => {
    test("Then it should always match this snapshot", () => {
      const testedRegisterPage = TestRenderer.create(
        <BrowserRouter>
          <ThemeProvider theme={theme}>
            <RegisterPage />
          </ThemeProvider>
        </BrowserRouter>
      );

      expect(testedRegisterPage).toMatchSnapshot();
    });

    test("Then it should render a register form with a button with the text 'Register'", () => {
      render(
        <BrowserRouter>
          <ThemeProvider theme={theme}>
            <RegisterPage />
          </ThemeProvider>
        </BrowserRouter>
      );

      const expectedButton = screen.getByRole("button", { name: "Register" });

      expect(expectedButton).toBeInTheDocument();
    });
  });
});
