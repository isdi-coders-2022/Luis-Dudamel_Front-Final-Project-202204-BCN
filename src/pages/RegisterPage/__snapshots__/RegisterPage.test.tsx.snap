// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a RegisterPage component page When it's invoked Then it should always match this snapshot 1`] = `
<div
  className="sc-bczRLJ hqXxVU"
>
  <main
    className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  >
    <div
      className="MuiBox-root css-htzt7r"
    >
      <div
        className="avatars-container"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular css-y2r78m-MuiAvatar-root"
        >
          <img
            className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="../../images/doberman.png"
          />
        </div>
        <div
          className="MuiAvatar-root MuiAvatar-circular css-y2r78m-MuiAvatar-root"
        >
          <img
            className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="../../images/dog.png"
          />
        </div>
        <div
          className="MuiAvatar-root MuiAvatar-circular css-y2r78m-MuiAvatar-root"
        >
          <img
            className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="../../images/greyhound.png"
          />
        </div>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h5 css-1uviv7h-MuiTypography-root"
      >
        Welcome to BarkedIn!
      </h1>
      <form
        className="MuiBox-root css-1y9cz7a"
        noValidate={true}
      >
        <div
          className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1gv81fz-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="username"
            id="username-label"
          >
            Username
            <span
              aria-hidden={true}
              className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-hiddenLabel css-1q5ijy-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={true}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputHiddenLabel css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="username"
              name="username"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Username
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1gv81fz-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="password"
            id="password-label"
          >
            Password
            <span
              aria-hidden={true}
              className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1q5ijy-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="current-password"
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="password"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Password
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1gv81fz-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="email"
            id="email-label"
          >
            Email
            <span
              aria-hidden={true}
              className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1q5ijy-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="email"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="email"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Email
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButtonBase-root  css-593tnw-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Register
        </button>
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <a
              className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-1ucbcvl-MuiTypography-root-MuiLink-root"
              href="#"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Already have an account? Log in here
            </a>
          </div>
        </div>
      </form>
    </div>
  </main>
</div>
`;
